src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -151,30 +151,13 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
         this.updateUiViewport();
 
         const {
-            style_as_list_view,
-            style_cell,
-            style_cell_conditional,
-            style_data,
-            style_data_conditional,
-            style_filter,
-            style_filter_conditional,
-            style_header,
-            style_header_conditional,
-            style_table
+            fixed_columns,
+            fixed_rows
         } = this.props;
 
-        this.handleResizeIf(
-            style_as_list_view,
-            style_cell,
-            style_cell_conditional,
-            style_data,
-            style_data_conditional,
-            style_filter,
-            style_filter_conditional,
-            style_header,
-            style_header_conditional,
-            style_table
-        );
+        if (fixed_columns || fixed_rows) {
+            this.handleResizeIf(...R.values(this.props));
```

Comment:
Don't attempt resize if nothing is fixed. Otherwise, resize on any prop change. Most props can require a recalculation due to interactions with other props. This will have a performance impact as resizing will happen more often, but the bulk of the cost (relayout) would be happening anyway.

Better to be right slightly slower than to be wrong slightly faster..